USER_GUIDE.md

```diff
@@ -478,6 +478,10 @@ print(response)
 
 Make sure to use `AsyncOpenSearch` with the `AsyncHttpConnection` connection class with the async `AWSV4SignerAsyncAuth` signer.
 
+```bash
+pip install opensearch-py[async]
+```
```

Comment:
Great. I think we can reorganize the guide a bit. We have sections like "Pre-requisites to use AWSV4SignerAuth", maybe you want to do a pass and consolidate these things into Setup? Just thinking out loud.

Reply:
Hello @dblock, take a look at this [draft PR](https://github.com/opensearch-project/opensearch-py/pull/340). Do I need to merge "Pre-requisites to use" into "Setup" and also do you want me to make any other changes.

Reply:
I commented on that PR. It's really NBD.